test/functional/entity-metadata/entity/Counters.ts

```diff
@@ -18,7 +18,7 @@ export class Counters {
     @Column()
     favorites: number;
 
-    @Column(() => Subcounters)
+    @Column(() => Subcounters, {prefix: "sub"})
```

Comment:
Why did you change exist test? You need to create a separate test for a new use case.

Reply:
The columns inside embedded entities were being incorrectly generated previously. The correct column name is `countersSubcountersWatchedUserId`, which is 32 characters long, and too long for Oracle DB which can only take 30 characters. Before this fix, the column name would have been `watchedUserId` or something, so there was no problem, but the wrong column name was going unnoticed.

This test only checks the ORM side of things anyway, so the exact name in database was irrelevant, but I can change the variable name from `subcounters` to `sub` directly if you'd prefer.  I have already created a new test case for the issue #6977.

Reply:
the problem is, if we merge these changes, we introduce a breaking change. Right now Oracle users have same case working (even if column name is invalid), but if we merge these changes their codebases won't be working at all. We can't tell them "hey guys just monkey patch it using "prefix" like we did it on our tests". Proper solution is to fix related Oracle issue.

Reply:
For this test anyway there is no solution for Oracle unless we plan to truncate column names automatically? Could potentially introduce a more informative error during column generation if target DB is Oracle since the DB error is very vague.

Reply:
we already do truncate in [some other places](https://github.com/typeorm/typeorm/blob/494866a999606e4acf8275aaae14099b0c1a0be1/src/find-options/FindOptionsUtils.ts#L229) for Oracle

Reply:
Might be something worth looking at with naming strategies for the future, gives the user the option for more meaningful change than simple truncation.